File: practice/dp/c++/1_maximum_sum.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int N = 3;
    vector<int> A = {7, -6, 9};

    vector<int> dp(N + 1, 0);

    for (int i = 0; i < A.size(); i++) {
        dp[i + 1] = max(dp[i], dp[i] + A[i]);
    }

    cout << dp[N] << endl;

    return 0;
}

File: practice/dp/rust/1_maximum_sum.rs
use std::cmp::max;

fn main() {
    let n: usize = 3;
    let a: Vec<i32> = vec![7, -6, 9];

    let mut dp: Vec<i32> = vec![0; n + 1];

    for (i, v) in a.iter().enumerate() {
        dp[i + 1] = max(dp[i], dp[i] + *v)
    }

    println!("{}", dp[n])
}
